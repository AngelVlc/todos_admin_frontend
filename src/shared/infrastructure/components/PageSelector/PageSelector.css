.pageNumber {
  width: 40px;
  text-align: right;
}