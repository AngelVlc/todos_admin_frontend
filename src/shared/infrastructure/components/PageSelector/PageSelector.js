import React from "react";
import "./PageSelector.css";

export const PageSelector = (props) => {
  const pageNumberChanged = (event) => {
    props.changePagination({
      ...props.paginationInfo,
      pageNumber: parseInt(event.target.value),
    });
  };

  const previousPageClicked = () => {
    props.changePagination({
      ...props.paginationInfo,
      pageNumber: props.paginationInfo.pageNumber - 1,
    });
  };

  const nextPageClicked = () => {
    props.changePagination({
      ...props.paginationInfo,
      pageNumber: props.paginationInfo.pageNumber + 1,
    });
  };

  return (
    <div className="ml-3 level-item">
      {props.paginationInfo.pageNumber > 1 && (
        <span className="icon" onClick={() => previousPageClicked()}>
          <i className="fas fa-thin fa-backward" />
        </span>
      )}
      <input
        type="number"
        min="1"
        value={props.paginationInfo.pageNumber}
        className="pageNumber"
        onChange={(e) => pageNumberChanged(e)}
        data-testid='pagenumber-input'
      ></input>
      <span className="icon" onClick={() => nextPageClicked()}>
        <i className="fas fa-thin fa-forward" />
      </span>
    </div>
  );
};
