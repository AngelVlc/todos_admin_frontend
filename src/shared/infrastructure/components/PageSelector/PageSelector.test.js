import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { PageSelector } from "./PageSelector";

describe("PageSelector", () => {
  const mockedChangePagination = jest.fn();

  const renderComponent = (paginationInfo) => {
    return {
      ...render(
        <PageSelector
          paginationInfo={paginationInfo}
          changePagination={mockedChangePagination}
        />
      ),
    };
  };

  afterEach(cleanup);

  it("Should match the snapshot", () => {
    const { asFragment } = renderComponent({
      pageNumber: 5,
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should change the page number", async () => {
    const { getByTestId } = renderComponent({
      pageNumber: 10,
    });

    await changeInputValue(getByTestId, "pagenumber-input", 5);

    expect(mockedChangePagination.mock.calls.length).toBe(1);
    expect(mockedChangePagination.mock.calls[0][0]).toStrictEqual({
      pageNumber: 5,
    });
  });
});

const changeInputValue = async (getByTestId, name, value) => {
  await waitFor(() => {
    fireEvent.change(getByTestId(name), {
      target: {
        value: value,
      },
    });
  });
};
