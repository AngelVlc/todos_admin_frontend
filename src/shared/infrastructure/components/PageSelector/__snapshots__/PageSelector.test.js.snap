// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSelector Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ml-3 level-item"
  >
    <span
      class="icon"
    >
      <i
        class="fas fa-thin fa-backward"
      />
    </span>
    <input
      class="pageNumber"
      data-testid="pagenumber-input"
      min="1"
      type="number"
      value="5"
    />
    <span
      class="icon"
    >
      <i
        class="fas fa-thin fa-forward"
      />
    </span>
  </div>
</DocumentFragment>
`;
